skglm/solvers/cd_solver.py

```diff
@@ -268,19 +269,27 @@ def cd_solver(
         for epoch in range(max_epochs):
             if is_sparse:
                 _cd_epoch_sparse(
-                    X.data, X.indptr, X.indices, y, w, Xw, datafit, penalty,
-                    ws)
+                    X.data, X.indptr, X.indices, y, w[:n_features], Xw,
+                    datafit, penalty, ws)
             else:
-                _cd_epoch(X, y, w, Xw, datafit, penalty, ws)
+                _cd_epoch(X, y, w[:n_features], Xw, datafit, penalty, ws)
+
+            # update intercept
+            if fit_intercept:
+                intercept_old = w[-1]
+                w[-1] -= datafit.intercept_update_step(y, Xw)
+                Xw += (w[-1] - intercept_old)
 
             # 3) do Anderson acceleration on smaller problem
             # TODO optimize computation using ws
-            w_acc[ws], Xw_acc[:], is_extrapolated = accelerator.extrapolate(w[ws], Xw)
+            w_acc[:], Xw_acc[:], is_extrapolated = accelerator.extrapolate(w, Xw)
 
             if is_extrapolated:  # avoid computing p_obj for un-extrapolated w, Xw
                 # TODO : manage penalty.value(w, ws) for weighted Lasso
-                p_obj = datafit.value(y, w, Xw) + penalty.value(w)
-                p_obj_acc = datafit.value(y, w_acc, Xw_acc) + penalty.value(w_acc)
+                p_obj = (datafit.value(y, w[:n_features], Xw) +
```

Comment:
Perhaps we should consider, in another PR removing the ``w`` argument in ``datafit.value``. It's pretty disturbing as ``.value`` does not use it. Also, in our setup we assume that datafit depends on ``Xw``. WDYT?

Reply:
I agree it makes sense at first, but unfortunately the `w` argument is needed for SVC